### DISCUSSION TITLE
WGS84 coordinates in accessors

### DISCUSSION THREAD

**Discussion by keschaefer** [visgl/deck.gl#5984] [Q&A] [answered]
Hi all- I am trying to apply styles to map features based on their coordinate data using turf.js. I'm doing some geometry checks, specifically, checking if a point, line, or polygon is contained within another polygon with `booleanContains`. In order to do this check, I first need to convert the point, line, or polygon to a turf.js friendly format using one of their helper functions (eg: `turf.polygon()`). My coordinates for the bounding shape are in WGS84 format and so I need my coordinates for each map feature I'm checking to be in WGS84 as well. 

I am able to pull the WGS84 coordinates off of map features via the `onClick` handler (https://deck.gl/docs/whats-new#features-in-wgs84-coordinates) but I'm hoping to get the coordinates in WGS84 format when using them in the `getFillColor` and `getLineColor` accessors in an MVTLayer specifically, because I need to map through all possible map features to do my booleanContains check to apply the appropriate styles. 

For example, for a polygon feature, I see something like these coordinates in the `getLineColor` accessor: 

```
geometry: {
 coordinates: [
   [-0.21701015625, -0.25],
   [-0.181884765625, -0.189453125],
   [-0.228271484375, -0.109375],
   [-0.25, -0.109375],
   [-0.25, -0.25],
   [-0.217041015625, -0.25]
 ]
}
```

which I understand to be a format that in some way references the tile it is within, like a relative position, but I would like for this data to be WGS84.

Wondering if there is:
a) a way to pass in an option to the MVTLayer to return the geometry coordinate data in WGS84 format in the accessors? (preferred)
b) a way to convert the above formatted data back to WGS84 that is performant, I work with datasets which can be hundreds of millions of rows, so lots of features to map through. This feels unlikely with such large datasets, but maybe?

**Comment by kylebarron** [comment, marked as answer]
This has been asked a few times before in issues; maybe try https://github.com/visgl/deck.gl/issues/4879#issuecomment-677771431?

**Reply by keschaefer** [reply]
Hello- thanks for this, yes, looks similar but I believe that post is related specifically to the data returned in the onClick and onHover. I was hoping to get it in a few other accessors which don't provide `pickingInfo`.

My suspicion is that with accessors that need to provide coordinates for all features in a layer, it has been deemed too expensive to provide the coordinates in the WGS84 format and thus we get back those relative coordinates instead.

**Reply by kylebarron** [reply]
The MVT Layer is designed to display MVT data as performantly as possible. Because of that there's no reprojection done to WGS84 when loading the data, because then the layer would need to reproject back to web mercator when rendering.

**Reply by Pessimistress** [reply]
It would be a lot more efficient if you perform the turf.js checks in local coordinates. You can convert a lnglat coordinate to the tile coordinate system like this:

```js
import {lngLatToWorld} from '@math.gl/web-mercator';

function wgs84CoordsToLocal(lngLat) {
  const webMercatorCoords = lngLatToWorld(lngLat);
  const scale = 2 ** tile.z / 512;
  return [
    webMercatorCoords[0] * scale - tile.x,
    (512 - webMercatorCoords[1]) * scale - tile.y
  ];
}
```
